### ISSUE TITLE
False positive on UnusedImport when using 'as' and the same package

### ISSUE THREAD

**Issue by mikezx6r** [arturbosch/detekt#1385]
## Expected Behavior
Using an import for a function, and renaming via 'as' should not report Unused unless the renamed value truly is not used

## Observed Behavior
False positive being reported in this scenario

## Steps to Reproduce
Create class that imports a function in the same package, but provides a different name

`import arrow.instances.traverse as eitherTraverse`

and ensure that eitherTraverse is referenced in the code

## Context
Many false positives so either need to turn this rule of OR add a file level @Suppres

## Your Environment
Detekt 1.0.0-RC12. Gradle 4.10. Windows 10. JDK8. Kotlin 1.3.0

**Comment by marianosimone** [comment]
There are actually unit tests to verify that aliases are respected for this rule. @mikezx6r do you have a minimal project that reproduces this problem?

**Comment by mikezx6r** [comment]
I don't currently have a minimal project, but ran into this when upgrading the version of Detekt used by the Arrow project. 

In this commit: https://github.com/arrow-kt/arrow/commit/97fcf188438c27ea44c840d112a6569a83220ca1, you'll see several places Suppressing UnusedImports.

If that's too much to look at, let me know, and I'll create a minimal project that exhibits this.

**Comment by TWiStErRob** [comment]
Run `gradlew detekt` on [detektUnusedImport.zip](https://github.com/arturbosch/detekt/files/2780021/detektUnusedImport.zip)
project.

Code:
```kotlin
package com.example

import java.util.Calendar // 3: truly unused
import com.example.foo as myFoo // 4: false positive
import com.example.other.foo as otherFoo // 5: used the same way as import alias above

fun f() : Boolean {
	return myFoo() == otherFoo()
}
```
Expected:
```
Ruleset: style - 10min debt
        UnusedImports - [Imports.kt] at detektUnusedImport\Imports.kt:3:1
```
Actual:
```
Ruleset: style - 10min debt
        UnusedImports - [Imports.kt] at detektUnusedImport\Imports.kt:3:1
        UnusedImports - [Imports.kt] at detektUnusedImport\Imports.kt:4:1
```

**Comment by TWiStErRob** [comment]
Introduced in #985?

**Comment by arturbosch** [comment]
Thank you for the code example @TWiStErRob . I could reproduce and fix it. Will be released in a new version later this weekend.
